Firestore/src/Query.php

```diff
@@ -1014,4 +978,144 @@ private function basePath()
             ? $this->parentName
             : $this->childPath($this->parentName, $this->query['from'][0]['collectionId']);
     }
+
+    /**
+     * Create a non composite Filter.
+     *
+     * @param string|array|FieldPath $fieldPath A field to filter by.
+     * @param string|int $operator An operator to filter by.
+     * @param mixed $value A value to compare to.
+     * @return array A non composite Filter array.
+     * @throws \InvalidArgumentException If an invalid operator or value is encountered.
+     */
+    private function createNonCompositeFilter($fieldPath, $operator, $value)
+    {
+        $basePath = $this->basePath();
+        if ($value instanceof FieldValueInterface) {
+            throw new \InvalidArgumentException(sprintf(
+                'Value cannot be a `%s` value.',
+                FieldValue::class
+            ));
+        }
+
+        if (!($fieldPath instanceof FieldPath)) {
+            $fieldPath = FieldPath::fromString($fieldPath);
+        }
+
+        $escapedPathString = $fieldPath->pathString();
+
+        // alias for friendlier error message below.
+        $originalOperator = $operator;
+        $operator = array_key_exists(strtolower($operator), $this->shortOperators)
+            ? $this->shortOperators[strtolower($operator)]
+            : $operator;
+
+        try {
+            FieldFilterOperator::name($operator);
+        } catch (\UnexpectedValueException $e) {
+            throw new \InvalidArgumentException(sprintf(
+                'Operator %s is not a valid operator',
+                $originalOperator
+            ));
+        }
+
+        if ($escapedPathString === self::DOCUMENT_ID) {
+            if ($operator === FieldFilterOperator::IN) {
+                $value = array_map(function ($value) use ($basePath) {
+                    return $this->createDocumentReference($basePath, $value);
+                }, (array) $value);
+            } else {
+                $value = $this->createDocumentReference($basePath, $value);
+            }
+        }
+
+        if ((is_float($value) && is_nan($value)) || is_null($value)) {
+            if ($operator !== FieldFilterOperator::EQUAL) {
+                throw new \InvalidArgumentException('Null and NaN are allowed only with operator EQUALS.');
+            }
```

Comment:
It seems this code pre-existed, but I am concerned this will _always_ throw an exception with a float value. 
Can we not compare float values using operators other than equal in Firestore?

Reply:
No, it will work fine. There's an `&&` in the first condition.